### PR TITLE
PageLayout: Use HTML5 landmark elements

### PR DESCRIPTION
**Pull request by colebemis**
Based on @alliethu's recommendation in her [accessibility review](https://github.com/github/primer/issues/782), the `PageLayout` component now renders [HTML5 landmark elements](https://web.dev/use-landmarks/) (`header`, `aside`, `footer`) to improve the navigation experience for people using assistive technologies (like screen readers). 

html5 landmarks in pagelayout component

### Merge checklist

- [x] Added/updated tests
- [ ] Added/updated documentation
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

File: src/PageLayout/PageLayout.tsx [hunk 1]
```diff
@@ -286,6 +286,7 @@ const Pane: React.FC<PageLayoutPaneProps> = ({
   const computedDividerWhenNarrow = dividerWhenNarrow === 'inherit' ? divider : dividerWhenNarrow
   return (
     <Box
+      as="aside"
```
Comment (colebemis) [inline src/PageLayout/PageLayout.tsx:289]: Waiting on an answer to this question: https://github.com/github/primer/issues/782#issuecomment-1069606557

File: src/PageLayout/PageLayout.tsx [hunk 2]
```diff
@@ -114,7 +114,6 @@ const HorizontalDivider: React.FC<DividerProps> = ({variant = 'none', variantWhe
   const {padding} = React.useContext(PageLayoutContext)
   return (
     <Box
-      role="separator"
```
Comment (pksjce) [inline src/PageLayout/PageLayout.tsx:117]: ooh interesting, what does removing this do?

**Review by rezrah — APPROVED** [review]
👍  nice

File: src/PageLayout/PageLayout.tsx (hunk 2 above)
Comment (siddharthkp) [reply src/PageLayout/PageLayout.tsx:117]: If doesn't have role= separator should this have aria-hidden=true instead?

**Review by siddharthkp — APPROVED** [review]
other than the Divider, everything looks good to me!

File: src/PageLayout/PageLayout.tsx (hunk 2 above)
Comment (alliethu) [reply src/PageLayout/PageLayout.tsx:117]: Good callout @siddharthkp. I believe my recommendation was to use both `role="presentation"` and `aria-hidden="true"`. @colebemis, I'll update the recommendation in the a11y review issue.

Comment (colebemis) [reply src/PageLayout/PageLayout.tsx:117]: Is this necessary to do on a `div` element?

Comment (colebemis) [reply src/PageLayout/PageLayout.tsx:117]: Without adding `role` or `aria-hidden`, it like looks divider already does not appear in the accessibility tree: 

CleanShot 2022-03-17 at 16 25 07@2x

What do `role` and `aria-hidden` do for us in this case?

Comment (siddharthkp) [reply src/PageLayout/PageLayout.tsx:117]: Ooh because it's an empty div, it would probably just get "Ignored" in the tree

**Pull request merged**